tests/test_fit_result.py

```diff
@@ -3,79 +3,106 @@
-     LeastSquares, (True, True, True, False, False))
-])
-def test_model(kwargs, objective_name, gof_presence):
-    model = Model({y: a * x ** b})
-    fit = Fit(model, **kwargs)
+@pytest.mark.parametrize('fit_kwargs, expected_par, expected_std, expected_obj, expected_got',
+    [
+        # No specific objective
+        (dict(x=np.linspace(1, 10, 10), y=3 * np.linspace(1, 10, 10) ** 2), {a: 3, b: 2}, 
+         {a: 0, b: 0}, LeastSquares, {'r_squared': 1.0, 'objective_value': 1e-23, 'chi_squared': 1e-23}),
+        # Test objective LeastSqueares
+        (dict(objective=LeastSquares, x=np.linspace(1, 10, 10), y=3 * np.linspace(1, 10, 10) ** 2), {a: 3, b: 2}, 
+         {a: 1e-13, b: 1e-13}, LeastSquares, {'r_squared': 1.0, 'objective_value': 1e-23, 'chi_squared': 1e-23}),
+        # Test objective LogLikelihood
+        (dict(objective=LogLikelihood, x=np.linspace(1, 10, 10)), {a: 62.56756, b: 758.08369}, 
+         {a: float('nan'), b: float('nan')}, LogLikelihood, 
+         {'objective_value': -float('inf'), 'log_likelihood': float('inf'), 'likelihood': float('inf')})
+    ])
+def test_no_constraint(minimizer, fit_kwargs, expected_par, expected_std, expected_obj, expected_got):
+
+    # Values of a and b by COBYLA and TrustConstr of (objective = LeastSquares, None)
+    # Values of a and b always of (objective = LogLikeihood)
+    # LBFGSB has no status message/not string
+    fit = Fit(**fit_kwargs, minimizer=minimizer, model=Model({y: a * x ** b}))
     fit_result = fit.execute()
+    _run_tests(fit_result, expected_par, expected_std, expected_obj, expected_got)
 
-    assert isinstance(fit_result.params, OrderedDict)
 
+@pytest.mark.parametrize('minimizer',
+    subclasses(ConstrainedMinimizer) - {ChainedMinimizer, DifferentialEvolution} | {None}
+    )
+@pytest.mark.parametrize('fit_kwargs, expected_par, expected_std, expected_obj, expected_got',
+    [
+        # No special objective
+        (dict(x=np.linspace(1, 10, 10), y=3 * np.linspace(1, 10, 10) ** 2), {a: 2.152889, b: 2.152889}, 
+         {a: 0.23715, b: 0.05076}, LeastSquares, {'r_squared': 0.99791, 'objective_value': 98.83587, 'chi_squared': 197.671746}),
+        # Test objective LeastSqueares
+        (dict(objective=LeastSquares, x=np.linspace(1, 10, 10), y=3 * np.linspace(1, 10, 10) ** 2), {a: 2.152889, b: 2.152889}, 
+         {a: 0.23715, b: 0.05076}, LeastSquares, {'r_squared': 0.99791, 'objective_value': 98.83587, 'chi_squared': 197.671746}),
+        # Test objective LogLikelihood
+        (dict(objective=LogLikelihood, x=np.linspace(1, 10, 10)), {a: 653.48460, b: 653.48460}, 
+         {a: float('nan'), b: float('nan')}, LogLikelihood, 
+         {'objective_value': -float('inf'), 'log_likelihood': float('inf'), 'likelihood': float('inf')})
+    ])
+def test_constraints(minimizer, fit_kwargs, expected_par, expected_std, expected_obj, expected_got):
+
+    # TrustConstr cannot handle LogLikelihood
```

Comment:
?!
That's a bug. Please elaborate.
Probably the LL objective doesn't have a Hessian?

Reply:
I would wait with it, because of another [discussion](https://github.com/tBuLi/symfit/pull/289#discussion_r450044360)
But the [error message](https://travis-ci.org/github/tBuLi/symfit/jobs/704984071#L1486) confirms your suspicion, that it doesn't have Hessian.

Reply:
Done #305

Reply:
please put a reference to the issue in a comment here as well

Reply:
In the code? :D

Reply:
Yes